deps/kms/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! # KMS
+//!
+//! This lib defines the API of the KMSes which will be used to implement
+//! image encryption and decryption.
+//!
+//! ## Trait Definitions
+//!
+//! ### Datakey
+//!
+//! [`Datakey`] is an abstraction of a key with the following attributes
+//! in different KMSes:
+//! - Can be used to encrypt/decrypt large pieces of data
+//! - Plaintext of the key can be exported outside the KMS
+//! - KMS can help to decrypt the ciphertext of the key to get the plaintext
+//! of the key.
```

Comment:
why do we need the KMS to "decrypt the ciphertext of the key", if "the plaintext of the key can be exported outside the KMS".

i thought keys were going to be wrapped using a TEE pubkey.

Reply:
This design philosophy comes from the API design from [aliyun KMS](https://www.alibabacloud.com/help/en/key-management-service/latest/kms-generatedatakey), [aws KMS](https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKey.html) etc. It is similiar to "Envelope encryption", s.t. wrap the datakey with CMK.